src/charm.py

```diff
@@ -96,17 +96,25 @@ def reconcile(self) -> None:
                 "Database configuration missing username, password or URI"
             )
             return
-        logger.info("Matrix Reconciling charm")
-        matrix = self._matrix.reconcile()
+        try:
+            logger.info("Matrix Reconciling charm")
+            matrix = self._matrix.get_matrix()
+            if matrix is None:
+                self.unit.status = ops.BlockedStatus("Matrix relation not found")
+                return
+        except ValidationError:
+            self.unit.status = ops.MaintenanceStatus("Matrix configuration not correct")
```

Comment:
Shouldnt the charm be blocked in this case too?

Reply:
There was a discussion somewhere with @gregory-schiano about this, if I recall correctly, we set Maintenance as the end user cannot do anything about it but wait - this assumes the data in the relation is not correct.

Reply:
I see. I was thinking about Maubot that an action would be removing the relation but its a different case.